### PR TITLE
chore: change log level to warning for empty keys

### PR DESCRIPTION
**Pull request by kostasrim**
* change log level to warning

### CODE REVIEW

File: src/server/rdb_load.cc [hunk 1]
```diff
@@ -2478,8 +2478,8 @@ void RdbLoader::LoadItemsBuffer(DbIndex db_ind, const ItemsBuf& ib) {
     PrimeValue pv;
     if (ec_ = FromOpaque(item->val, &pv); ec_) {
       if ((*ec_).value() == errc::empty_key) {
-        LOG(ERROR) << "Found empty key: " << item->key << " in DB " << db_ind << " rdb_type "
-                   << item->val.rdb_type;
+        LOG(WARNING) << "Found empty key: " << item->key << " in DB " << db_ind << " rdb_type "
```
[the remark this answers is not in the document]
Comment (adiholden) [reply src/server/rdb_load.cc:2481]: I think that we might have a bug that we saw empty keys not for set and hash set and we want to track this
am I right @chakaz

[the remark this answers is not in the document]
Comment (kostasrim) [reply src/server/rdb_load.cc:2481]: it was a hash set I think @adiholden but let's wait for @chakaz

[the remark this answers is not in the document]
Comment (chakaz) [reply src/server/rdb_load.cc:2481]: We definitely saw this line being printed for other types:
```
#define RDB_TYPE_HASH_ZIPLIST  13
#define RDB_TYPE_STRING 0
#define RDB_TYPE_SET    2
```

**Review by chakaz — DISMISSED** [review]

File: src/server/rdb_load.cc [hunk 2]
```diff
@@ -228,6 +228,11 @@ string ModuleTypeName(uint64_t module_id) {
   return string{name};
 }
 
+bool RdbTypeAllowedEmpty(int type) {
+  return type == RDB_TYPE_STRING || type == RDB_TYPE_JSON || type == RDB_TYPE_SBF ||
```
Comment (adiholden) [inline src/server/rdb_load.cc:232]: do we allow empty for this types?
I think this should be only 
constexpr uint8_t RDB_TYPE_HASH_WITH_EXPIRY = 31;
constexpr uint8_t RDB_TYPE_SET_WITH_EXPIRY = 32;
No?

Comment (kostasrim) [reply src/server/rdb_load.cc:232]: nope, we do allow. For strings: `SET foo ""` --> empty and for SBF we have a test :)

See `PrimeValue::TagAllowsEmptyValue`

File: src/core/compact_object.cc [hunk 3]
```diff
@@ -852,7 +852,8 @@ bool CompactObj::HasAllocated() const {
 
 bool CompactObj::TagAllowsEmptyValue() const {
   const auto type = ObjType();
-  return type == OBJ_JSON || type == OBJ_STREAM || type == OBJ_STRING || type == OBJ_SBF;
+  return type == OBJ_JSON || type == OBJ_STREAM || type == OBJ_STRING || type == OBJ_SBF ||
+         type == OBJ_SET;
```
Comment (kostasrim) [inline src/core/compact_object.cc:856]: We also allow `SETS` now since they can be encoded as dense sets

File: src/server/rdb_load.cc (hunk 2 above)
Comment (kostasrim) [reply src/server/rdb_load.cc:232]: synced with Adi internally, we decided to also not log as errors `hash sets and sets` with expiry

**Review by adiholden — APPROVED** [review]

**Pull request merged**